pool/pool.go

```diff
@@ -436,7 +446,9 @@ func (c *clientWrapper) objectSearch(ctx context.Context, prm PrmObjectSearch) (
 	if err != nil {
 		return nil, fmt.Errorf("init object searching on client: %w", err)
 	}
-	res.UseKey(*prm.key)
+	if prm.key != nil {
```

Comment:
Do we need `nil` check there? I mean this key _must_ be set, right? And if it isn't, then it is okay to panic and fix this issue.

Reply:
I'm not sure this key `must` be set. https://github.com/nspcc-dev/neofs-sdk-go/blob/master/client/object_search.go#L101-L102